src/api/providers/cline.ts

```diff
@@ -1,143 +1,233 @@
 import { Anthropic } from "@anthropic-ai/sdk"
-import OpenAI from "openai"
 import { ApiHandler } from "../"
+import { ClineAccountService } from "@/services/account/ClineAccountService"
 import { ApiHandlerOptions, ModelInfo, openRouterDefaultModelId, openRouterDefaultModelInfo } from "@shared/api"
 import { createOpenRouterStream } from "../transform/openrouter-stream"
 import { ApiStream, ApiStreamUsageChunk } from "../transform/stream"
-import axios from "axios"
+import axios, { AxiosRequestConfig, AxiosResponse } from "axios"
 import { OpenRouterErrorResponse } from "./types"
 import { withRetry } from "../retry"
+import { AuthService } from "@/services/auth/AuthService"
 
 export class ClineHandler implements ApiHandler {
 	private options: ApiHandlerOptions
-	private client: OpenAI
+	private clineAccountService = ClineAccountService.getInstance()
+	private _authService: AuthService
+	// TODO: replace this with a global API Host
+	private readonly _baseUrl = "https://core-api.staging.int.cline.bot"
+	// private readonly _baseUrl = "http://localhost:7777"
 	lastGenerationId?: string
+	private counter = 0
 
 	constructor(options: ApiHandlerOptions) {
 		this.options = options
-		this.client = new OpenAI({
-			baseURL: "https://api.cline.bot/v1",
-			apiKey: this.options.clineApiKey || "",
-			defaultHeaders: {
-				"HTTP-Referer": "https://cline.bot", // Optional, for including your app on cline.bot rankings.
-				"X-Title": "Cline", // Optional. Shows in rankings on cline.bot.
-				"X-Task-ID": this.options.taskId || "", // Include the task ID in the request headers
-			},
-		})
+		this._authService = AuthService.getInstance()
 	}
 
 	@withRetry()
 	async *createMessage(systemPrompt: string, messages: Anthropic.Messages.MessageParam[]): ApiStream {
+		const clineAccountAuthToken = await this._authService.getAuthToken()
+
 		this.lastGenerationId = undefined
 
-		const stream = await createOpenRouterStream(
-			this.client,
-			systemPrompt,
-			messages,
-			this.getModel(),
-			this.options.reasoningEffort,
-			this.options.thinkingBudgetTokens,
-			this.options.openRouterProviderSorting,
+		const requestConfig: AxiosRequestConfig = {
+			headers: {
+				"HTTP-Referer": "https://cline.bot", // Optional, for including your app on cline.bot rankings.
```

Comment:
gotta make sure we include this for as long as possible to proxy this to our openrouter calls and contribute to the leaderboards. just an aside, feel free to close after reading